### PR TITLE
refactor: run instance-manager workflow on PR for feature branches

### PR DESCRIPTION
**Pull request by radnov**
This PR enables running the updated [Build, test and deploy workflow](https://github.com/dhis2-sre/gha-workflows/pull/29) on PRs and only deploy when the `deploy` label is present (either initially added when creating the PR or after the PR was already created), as well as adds a new [Delete environment](https://github.com/dhis2-sre/gha-workflows/pull/29/files) that deletes a deployed environment when the PR is closed/merged. Note that the behaviour of the "Build, test and deploy" workflow remains the same for pushes to `master`, as well as creating new tags.

See https://github.com/dhis2-sre/im-manager/pull/318 for more details.

### CODE REVIEW

File: helm/chart/values.yaml [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 replicaCount: 1
 
-deployableNamespaces: []
+deployableNamespaces:
+  - whoami
```
Comment (tonsV2) [inline helm/chart/values.yaml:8]: Shouldn't we keep this as is and only overwrite using value files or the skaffold file?

File: helm/data/values/prod/values.yaml [hunk 2]
```diff
@@ -1,12 +1,6 @@
-imagePullPolicy: IfNotPresent
-
 deployableNamespaces:
   - whoami
```
Comment (tonsV2) [inline helm/data/values/prod/values.yaml:2]: I suggest we remove whoami from prod. If we as developers needs to test we can use dev

File: helm/data/values/prod/values.yaml [hunk 3]
```diff
@@ -1,12 +1,6 @@
-imagePullPolicy: IfNotPresent
```
Comment (tonsV2) [inline helm/data/values/prod/values.yaml:1]: Why is this removed?

File: helm/chart/values.yaml (hunk 1 above)
Comment (radnov) [reply helm/chart/values.yaml:8]: We have [the `whoami` group as a default value in the chart for the `im-manager`](https://github.com/dhis2-sre/im-manager/blob/master/helm/chart/values.yaml#L12-L14), so I thought it might be nice to keep it consistent for the inspector, too.

File: helm/data/values/prod/values.yaml (hunk 3 above)
Comment (radnov) [reply helm/data/values/prod/values.yaml:1]: This is [the default value in the chart](https://github.com/dhis2-sre/im-inspector/blob/master/helm/chart/values.yaml#L18) already, so I'm just making it consistent with the `im-manager` helm setup.

File: helm/data/values/prod/values.yaml (hunk 2 above)
Comment (radnov) [reply helm/data/values/prod/values.yaml:2]: Again, just for consistency with the `im-manager`, [where we have the `whoami` group](https://github.com/dhis2-sre/im-manager/blob/master/helm/data/values/prod/values.yaml#L1-L3).
